cookbook.go

```diff
@@ -164,3 +186,200 @@ func (c *CookbookService) Delete(name, version string) (err error) {
+	}
+
+}
+func trimQuotes(s string) string {
+	if len(s) >= 2 {
+		if c := s[len(s)-1]; s[0] == c && (c == '"' || c == '\'') {
+			return s[1 : len(s)-1]
+		}
+	}
+	return s
+}
+func getKeyValue(str string) (string, []string) {
+	c := strings.Split(str, " ")
+	if len(c) == 0 {
+		return "", nil
+	}
+	return strings.TrimSpace(c[0]), c[1:]
+}
+func isFileExists(name string) bool {
+	if _, err := os.Stat(name); errors.Is(err, os.ErrNotExist) {
+		return false
+	}
+	return true
+}
+
+func clearWhiteSpace(s []string) (result []string) {
+	for _, i := range s {
+		if len(i) > 0 {
+			result = append(result, i)
+		}
+	}
+	return result
+}
+
+func NewMetaData(data string) (m CookbookMeta, err error) {
+	linesData := strings.Split(data, "\n")
+	if len(linesData) < 3 {
+		return m, errors.New("not much info")
+	}
+	m.Depends = make(map[string]string, 1)
+	m.Platforms = make(map[string]interface{}, 1)
+	for _, i := range linesData {
+		key, value := getKeyValue(strings.TrimSpace(i))
+		if fn, ok := metaRegistry[key]; ok {
+			fn(value, &m)
+		}
+	}
+	return m, err
+}
+
+func NewMetaDataFromJson(data []byte) (m CookbookMeta, err error) {
+	err = json.Unmarshal(data, &m)
+	return m, err
+}
+
+func StringParserForMeta(s []string) string {
+	str := strings.Join(s, " ")
+	return trimQuotes(strings.TrimSpace(str))
+}
+func metaNameParser(s []string, m *CookbookMeta) {
+	m.Name = StringParserForMeta(s)
+}
+func metaMaintainerParser(s []string, m *CookbookMeta) {
+	m.Maintainer = StringParserForMeta(s)
+}
+func metaMaintainerMailParser(s []string, m *CookbookMeta) {
+	m.MaintainerEmail = StringParserForMeta(s)
+}
+func metaLicenseParser(s []string, m *CookbookMeta) {
+	m.License = StringParserForMeta(s)
+}
+func metaDescriptionParser(s []string, m *CookbookMeta) {
+	m.Description = StringParserForMeta(s)
+}
+func metaLongDescriptionParser(s []string, m *CookbookMeta) {
+	m.LongDescription = StringParserForMeta(s)
+}
+func metaIssueUrlParser(s []string, m *CookbookMeta) {
+	m.IssueUrl = StringParserForMeta(s)
+}
+func metaSourceUrlParser(s []string, m *CookbookMeta) {
+	m.SourceUrl = StringParserForMeta(s)
+}
+func metaGemParser(s []string, m *CookbookMeta) {
+	m.Gems = append(m.Gems, StringParserForMeta(s))
+}
+
+func metaVersionParser(s []string, m *CookbookMeta) {
+	m.Version = StringParserForMeta(s)
+}
+func metaOhaiVersionParser(s []string, m *CookbookMeta) {
+	m.OhaiVersion = StringParserForMeta(s)
+}
+func metaChefVersionParser(s []string, m *CookbookMeta) {
+	m.ChefVersion = StringParserForMeta(s)
+}
+func metaPrivacyParser(s []string, m *CookbookMeta) {
+	if s[0] == "true" {
+		m.Privacy = true
+	}
+}
+func metaSupportsParser(s []string, m *CookbookMeta) {
+	s = clearWhiteSpace(s)
+	switch len(s) {
+	case 1:
+		if s[0] != "os" {
+			m.Platforms[strings.TrimSpace(s[0])] = ">= 0.0.0"
+		}
+	case 2:
+		m.Platforms[strings.TrimSpace(s[0])] = s[1]
+	case 3:
+		v := trimQuotes(s[1] + " " + s[2])
+		m.Platforms[strings.TrimSpace(s[0])] = v
+
+	}
+	if len(s) > 3 {
```

Comment:
This should be returned to the caller as an error instead of aborting the call w/ output in the library.  The same applies in the other `panic` case below.